pkg/controller/worker/machines.go

```diff
@@ -179,7 +179,14 @@ func (w *workerDelegate) generateMachineConfig(_ context.Context) error {
 				"description":        fmt.Sprintf("Machine of Shoot %s created by machine-controller-manager.", w.worker.Name),
 				"disks":              disks,
 				"labels":             gceInstanceLabels,
-				"machineType":        pool.MachineType,
+				// TODO: make this configurable for the user
```

Comment:
Is there a concrete need already known to let users pass additional metadata for the machines?
If yes, we might wanna create an issue for it as it is better trackable?

Reply:
In the current `mcm-provider-gcp` code, we have a feature to add `metadata` field on a VM instance at the time of its creation(See [this](https://github.com/gardener/machine-controller-manager-provider-gcp/blob/ae925290549ce25b0116a5751ab3684a3d1a60b5/pkg/gcp/machine_controller_util.go#L129-L140)). But at the moment we cannot specify metadata field in the `providerConfig` section of the worker pool in the `shoot` yaml as the current `extension-provider-gcp` cannot decode it.(See [this](https://github.com/gardener/gardener-extension-provider-gcp/blob/3db08ee1f0c7ac279e47124360feeee841752675/pkg/apis/gcp/types_worker.go#L25) and [this](https://github.com/gardener/gardener-extension-provider-gcp/blob/3db08ee1f0c7ac279e47124360feeee841752675/pkg/controller/worker/machines.go#L137-L142)). So even though mcm provides the feature of specifying metadata on a VM instance, it will not be used as the machine class will never have this field present. This is the reason why I added this comment for now.

I am not opening an issue right now because we are unsure if we want to keep this field in the machine class. The reason is that any change in the `providerConfig` part of the `worker` section in the `shoot` yaml will generate a new machine class name (See [this](https://github.com/gardener/gardener-extension-provider-gcp/blob/3db08ee1f0c7ac279e47124360feeee841752675/pkg/controller/worker/machines.go#L206) and [this](https://github.com/gardener/gardener-extension-provider-gcp/blob/3db08ee1f0c7ac279e47124360feeee841752675/vendor/github.com/gardener/gardener/extensions/pkg/controller/worker/machines.go#L122)) and therefore a new machine class which will cause a rolling update of the machines. We don't want this just because some `metadata` field was changed. We already have an issue on mcm (See [this](https://github.com/gardener/machine-controller-manager/issues/750)) which requires something similar and should be sufficient to keep track of this as well.

cc @himanshu-kun